astropy/units/quantity_helper/erfa.py

```diff
@@ -98,6 +99,24 @@ def helper_gc2gd(f, nounit, unit1):
         )
 
 
+def helper_gc2gde(f, unit_r, unit_flat, unit_xyz):
+    from astropy.units.si import m, radian
+
+    if unit_r is None:
+        raise UnitTypeError("Equatorial radius should have length units.")
+    unit_flat = unit_r / unit_r
```

Comment:
Why is this necessary? You are overriding the input unit (which might be, say, `u.percent`, which could be converted to `dimensionless_unscaled`)

Reply:
ok, understood: this was the reason of the error on `unit_r` being managed separately.
I'm going to remove it then the check on `unit_r` will go too.